nrf802154_sniffer/nrf802154_sniffer.py

```diff
@@ -235,109 +237,100 @@ def control_reader(self, fifo):
         Thread responsible for reading wireshark commands (read from fifo).
         Related to not-yet-implemented wireshark toolbar features.
         """
-        with open(fifo, 'rb', 0 ) as fn:
+        with open(fifo, 'rb', 0) as fn:
             arg = 0
             while arg != None:
                 arg, typ, payload = Nrf802154Sniffer.control_read(fn)
             self.stop_sig_handler()
 
-    def control_writer(self, fifo, queue):
-        """
-        Thread responsible for sending wireshark commands (read from fifo).
-        Related to not-yet-implemented wireshark toolbar features.
-        """
-        with open(fifo, 'wb', 0 ) as fn:
-            while self.running.is_set():
-                time.sleep(1)
-
-    def serial_write(self, ser):
+    def serial_write(self):
         """
-        Function responsible for sending commands to serial port
+        Function responsible for sending commands to serial port.
         """
         command = self.serial_queue.get(block=True, timeout=1)
         try:
-            ser.write(command + b'\r\n')
-            ser.write(b'\r\n')
+            self.serial.write(command + b'\r\n')
+            self.serial.write(b'\r\n')
         except IOError:
             self.logger.error("Cannot write to {}".format(self))
             self.running.clear()
 
-    def serial_writer(self, ser):
+    def serial_writer(self):
         """
-        Thread responsible for sending commands to serial port
+        Thread responsible for sending commands to serial port.
         """
         while self.running.is_set():
             try:
-                self.serial_write(ser)
+                self.serial_write()
             except Queue.Empty:
                 pass
 
-        # Write final commands and break out
+        # Write final commands and break out.
         while True:
             try:
-                self.serial_write(ser)
+                self.serial_write()
             except Queue.Empty:
                 break
 
     def serial_reader(self, dev, channel, queue):
         """
         Thread responsible for reading from serial port, parsing the output and storing parsed packets into queue.
         """
-        # Wireshark needs this sleep for reset purposes
-        time.sleep(2)
```

Comment:
Are you sure it is safe to remove this sleep? There was a race condition during clicking "reset" button (or "stop" followed by "start", I'm not sure) in wireshark while there was intensive traffic captured by the sniffer. Sometimes sniffer reported very old frames as new one (flushing some buffer or something like this). This sleep gives sniffer time to flush buffers before it starts capturing new frames.

Reply:
I didn't observe this. However i used data jammer on second end so I might just not distinguish between old and new frames. I'm fine with returning sleep back. My observation was that it is needed to have sleep ~1-2s on Linux and ~2-5s on Windows to be able to push reset button, else serial port couldn't be opened, so this PR actually fixes this unwanted behavior.

Reply:
This is worth testing. IMO we can make the sniffer clear the receive buffer after the init commands are sent to circumvent this.

Reply:
It could be possible to clear buffers with init commands. On the other hand, the data could be present in UART buffers in USB or OS and it cannot be cleared by firmware. I agree, it should be tested.

Reply:
I reverted the previous behaviour, although i failed to reproduce the problem mentioned at the beginning.